app/Http/Controllers/Backend/Export/ExportController.php

```diff
@@ -19,15 +20,26 @@
 abstract class ExportController extends Controller
 {
 
+    /**
+     * @throws DataOverflowException If too many results are given.
+     */
     public static function getExportableStatuses(User $user, Carbon $timestampFrom, Carbon $timestampTo): Collection {
-        return Status::join('train_checkins', 'statuses.id', '=', 'train_checkins.status_id')
-                     ->where('statuses.user_id', $user->id)
-                     ->where('train_checkins.departure', '>=', $timestampFrom->startOfDay()->toIso8601String())
-                     ->where('train_checkins.departure', '<=', $timestampTo->endOfDay()->toIso8601String())
-                     ->select(['statuses.*'])
-                     ->get();
+        $statuses = Status::join('train_checkins', 'statuses.id', '=', 'train_checkins.status_id')
+                          ->where('statuses.user_id', $user->id)
+                          ->where('train_checkins.departure', '>=', $timestampFrom->startOfDay()->toIso8601String())
+                          ->where('train_checkins.departure', '<=', $timestampTo->endOfDay()->toIso8601String())
+                          ->select(['statuses.*'])
+                          ->limit(2001)
+                          ->get();
+        if ($statuses->count() >= 2001) {
```

Comment:
Der Code ist mir nicht so richtig verständlich auf dem ersten Blick, und könnte einen Kommentar vertragen. Sag mal bitte, was du davon hältst, damit wir auch in 3 Jahren noch verstehen, was der Code machen soll:

```suggestion
        // A user should only be able to export 2000 statuses at once to avoid memory
        // overflows. Thus, if the database returns 2001 entries (which is the limit),
        // there are `>2000` statuses in this time frame and the user must choose a
        // smaller time frame.
        if ($statuses->count() == 2001) {
```